### PR TITLE
Fix circular reference between CBLUnsavedRevision and CBLAttachment

### PR DESCRIPTION
**Pull request by pasin**
- As CBLUnsavedRevision keeps CBLAttachment objects in an array so setting the revision (self) object to the CBLAttachment objects will cause circular reference memory leak.

- Instead of setting the revision (self) object when adding a CBLAttachment to the CBLUnsavedRevision object, overiding the attachmentNamed: method and creating a new CBLAttachment object and adding a revision object there.

### CODE REVIEW

File: Source/API/CBLRevision.m [hunk 1]
```diff
@@ -367,12 +367,28 @@ - (void) _addAttachment: (CBLAttachment*)attachment named: (NSString*)name {
...
+    // [2] Setting the UnsavedRevision object here will cause the circular reference memory leak.
+    // The revision object will be set in the overridden attachmentName: method.
 }
 
 - (void) removeAttachmentNamed: (NSString*)name {
     [self _addAttachment: nil named: name];
 }
 
+- (CBLAttachment*) attachmentNamed: (NSString*)name {
+    CBLAttachment* base = self.attachmentMetadata[name];
```
Comment (snej) [inline Source/API/CBLRevision.m]: The object might be an NSDictionary not a CBLAttachment, if the attachment exists in the parent revision. (Look at the superclass method.)

Comment (snej) [inline Source/API/CBLRevision.m]: I actually don't think you need this override at all. The superclass method looks fine for this case.

Comment (pasin) [inline Source/API/CBLRevision.m]: I looked at the super class but as I didn't set the the revision to the attachment object, the super class will return the same CBLAttachment without revision object set.

Comment (pasin) [inline Source/API/CBLRevision.m]: For your 1st comment, I will need to handle that case.

**Comment by pasin** [comment]
@snej I updated the code and unit test. Please review them when you have time.

**Pull request merged**